src/hackney_request.erl

```diff
@@ -348,13 +353,6 @@ handle_body(Headers, ReqType0, Body0, Client) ->
                             S = hackney_headers_new:get_value(<<"content-length">>, Headers),
                             {S, CT, Body0};
 
-                          _ when is_list(Body0) -> % iolist case
```

Comment:
This branch isn't needed anymore since an iolist body will now be converted a to binary before being passed to `handle_body`.